pkg/auth/providers/saml/saml_client.go

```diff
@@ -262,6 +303,59 @@ func (s *Provider) getSamlPrincipals(config *v32.SamlConfig, samlData map[string
 	return userPrincipal, groupPrincipals, nil
 }
 
+// FinalizeSamlLogout processes the logout obtained by the POST to /saml/slo from IdP
+// ATTENTION: Expects to be called after the logout response was validated.
+// See `ServeHTTP`, saml_handlers.go
+func (s *Provider) FinalizeSamlLogout(w http.ResponseWriter, r *http.Request) {
+
+	if relayState := r.Form.Get("RelayState"); relayState != "" {
+		// delete the cookie
+		s.clientState.DeleteState(w, r, relayState)
+	}
+
+	redirectURL := s.clientState.GetState(r, "Rancher_FinalRedirectURL")
+
+	s.clientState.DeleteState(w, r, "Rancher_UserID")
+	s.clientState.DeleteState(w, r, "Rancher_Action")
+	s.clientState.DeleteState(w, r, "Rancher_FinalRedirectURL")
+
+	r.ParseForm()
+	err := s.serviceProvider.ValidateLogoutResponseRequest(r)
+	// -- ATTENTION -- ATTENTION -- ATTENTION -- ATTENTION --
+	// -- ATTENTION -- ATTENTION -- ATTENTION -- ATTENTION --
+	//
+	// Using crewjam/saml here requires a locally patched package
+	// See https://github.com/crewjam/saml/pull/491/files
+	// for patch - Fixes attempt to parse compressed data as XML
+	//
+	// TODO : Fork !!
```

Comment:
Has this fork been done yet?  i.e. Can we remove the TODO?

Reply:
yes, fork is done, todo can go.

Reply:
Done with commit [7437e4747656226f7291311d9ae2fa4b9ff79e58]